Adding content to the search page documentation

jackrechsteiner opened the pull request:
The "Searching the corpus" documentation page can still use some refinement, but it's gotten to a point were it would be useful to get some feedback on it.

jackrechsteiner committed:
Adding content to search page documentation

jackrechsteiner committed:
Adding search progress bar section, updating sections that had unclear information, and adding in more try-its

jackrechsteiner committed:
Editing wording and formatting, adding in a "what you can do" section, and adding more try its

jackrechsteiner committed:
Merge remote-tracking branch 'upstream/main'

jackrechsteiner committed:
Initial very rough draft in progress

jackrechsteiner committed:
Reorganizing some sections, adding in new sections, and fleshing out the regex demo section

jackrechsteiner committed:
Verbifying section names and updating relative links

jackrechsteiner committed:
Correcting section name and link that was missed before

jackrechsteiner committed:
Changing "layers with specific notation systems" to "Searching phonological and syntactic layers", adding screengrabs and try-its for phono+syntax section

jackrechsteiner committed:
Merge remote-tracking branch 'upstream/main'

jackrechsteiner committed:
Restructuring phonological and syntactic layers section

jackrechsteiner committed:
Mostly filling out the "Searching segments within words" section, but also moving the "doesn't match" try-it back to where it was originally and cleaning up some commented notes

jackrechsteiner commented:
The search page is still a work in progress, but I've been going through the sections in a linear order so I wanted to push the updates that I have been able to do so they can be reviewed while the other sections are still being refined.

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+1. [Match and display options](#match-and-display-options)
+1. [Search progress bar](#search-progress-bar)
+
+![]({{ '/assets/screengrab/search/search-page-layout.png' | relative_url }})
```
I think in general the first screengrab on the how-to pages should be what users first see when they load the page, without selecting any options, expanding any menus, entering any input, etc. (The search results page is an obvious exception!) Here, we can have the current screengrab come second to display the progress bar.

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
 It all starts from the _search_ page: <https://apls.pitt.edu/labbcat/search>.
 Once you create a search, you can [export the results]({{ '/doc/export-data' | relative_url }}) in a variety of formats.
 
+[^patterns]: The term "patterns" is used throughout this page because the search function is capable of matching exact words, affixes, individual sound segments, stress, part of speech, and much more!
```
Italics on patterns

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+
+![]({{ '/assets/screengrab/search/search-page-layout.png' | relative_url }})
 
 When you first load _search_, you'll only see the first three;
```
I know we've been going with shorter page names in general, but to reduce ambiguity with the button name, let's go with `the _search_ page`

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+There are two ways to select participants for this filter on the _search_ page:
+
+- Clicking _Participants_ will take you to the [_participants_ page]({{ '/doc/browse-participants' | relative_url }}).
+  From this page, click the checkboxes for all participants you would like to include in your search and then click the _Layered Search_ button to return to the _search_ page.
```
Re: "click the checkboxes", if you specify a filter and don't click any checkboxes, then it'll select all participants in the filter.

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+    Additionally, the `.csv` upload option treats multiple rows of the same participant code as if they were a single row.
+    If there is an error in the column, it will be ignored (e.g., if the column contains `CB96` instead of `CB06`, the filter will simply ignore `CB96` because it is not a valid participant code).
+
+<!-- this is now described on the new apls-dev search page -->
```
It'll be good to have some of this clarification in here all the same!

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+    If there is an error in the column, it will be ignored (e.g., if the column contains `CB96` instead of `CB06`, the filter will simply ignore `CB96` because it is not a valid participant code).
+
+<!-- this is now described on the new apls-dev search page -->
+
```
Let's add a try-it here

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+
+{:.note}
+> If you've selected participants for the participants filter, only the transcripts for those participants will be displayed when using the transcripts filter.
+> Additionally, if you select transcripts before selecting participants, then selected participants will cause your transcripts filter to be overwritten.
```
Change to "then your transcripts filter will be overwritten"

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+Selecting a layer in the layer selection menu will make an input field for that layer appear in the pattern input fields section.
+
+{:.note}
+> Layers that are selected with the layer selection menu are only applied to the search if input is entered into their pattern input fields.
```
Let's scoot this down lower so we can get to users trying their first search quicker. (Also because this is more relevant to multi-layer searches.)

In fact, we might consider breaking with format and having a "Your first search" section above. Or having search subpages, which would allow us to have a search 'splash' page where a "Your first search" makes sense. (I also recognize that the old format of the _search_ page lends itself to not-the-most-ideal ordering because it invites you to look at the participant/transcript filters first--I'm hoping I'll be able to push UI updates ASAP.)

djvill replied:
Also, let's switch to active voice to make it a little more accessible in style: "If you select a layer but don't enter anything into its pattern input field, it doesn't get applied to the search"

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+{:.note}
+> Layers that are selected with the layer selection menu are only applied to the search if input is entered into their pattern input fields.
+
+<!-- the apls-dev search page now has more layer icons and the option to hide layer icons -->
```
In the new UI, the layer picker is the same* used on the _transcript_ page, so we'll be able to introduce it there and refer back to it here. 

\* With some minor differences. As of right now:
| Transcript | Search |
|------------|---------|
| Layers aren't selectable if they're empty in that transcript | No such thing as empty layers, so all layers selectable |
| Layers turn fun colors when selected | No fun colors (though that's easy to add if that'd be a useful change) |
| turn, utterance not selectable | all selectable |
| word always selected (and currently not shown in the layer picker, tho I intend to change that) | word not always selected |
| Phonological-type layers get an option to toggle between IPA & DISC labels | No such option |

jackrechsteiner replied:
In line with this, I've also updated the terminology on the 'Searching the corpus' doc page to be "layer selector" to maintain consistency with the 'Viewing a transcript' doc page

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+
+The pattern input fields enable you to specify the pattern you'd like to search for.
+
+{:.note .no-collapse}
```
Let's move this lower. I worry that if we focus too much early on on what users can't do or what will mess things up, it'll put new/potential users in a defensive/scared headspace.

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+The _matches_ drop-down menu is located below the layer name of the input field.
+This drop-down menu lets you select whether the search will identify annotations that match your input pattern or annotations that don't match your input pattern.
+
+{: .try-it }
+> To find all matches for the segment /u/ that don't have primary stress:
+> 1. Go to the [_search_ page](https://apls.pitt.edu/labbcat/search).
+> 1. Click the checkbox for the <span class="layer">segment</span> layer to make the segment input field appear.
+> 1. Enter `u` into the segment input field.
+> 1. Select the `phonology` project in the layer selection menu and click the checkbox for the <span class="layer">stress</span> layer to make the stress input field appear.
+> 1. Click the drop-down selection menu on the right side of the stress input field and select _primary_.
+> 1. Click the _matches_ drop-down menu for the stress input field and select _doesn't match_.
+> 1. Click the _Search_ button.
```
Whoa, this is biting off way more than new users can chew. Let's start with an orthography try-it: How many times is X word in the corpus? Y word? One possible option is `pirates` vs. `steelers` vs. `penguins`, as it's accessible and could lead nicely into the regex search (e.g., `pirates?`, `pirate|steeler|penguin`); we would have to mention right away that the orthography layer is all lowercase, which might or might not be a drawback.

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+  > 1. Click the _Search_ button.
+
+{:.note}
+> It is recommended to use the APLS search [anchor options](#anchoring-searches) instead of the regex anchors `^` and `$` because not all layers support anchoring and `$` is the DISC symbol for /ɔ/.
```
Let's change this to stg like "If you're familiar with regular expressions, then you know that `^` and `$` usually anchor matches to the start and end of the pattern. These aren't necessary in APLS because regular expressions always match the entire annotation. This is why if you want words that start with a string, you need to use `.*` at the end. [line break] A separate but related topic is anchoring a search match to a larger annotation, such as finding words at the start of a turn. This is covered below."

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+  > To search for every instance of "breathe" and "breath":
+  > 1. Go to the [_search_ page](https://apls.pitt.edu/labbcat/search).
+  > 1. Enter `breath(e)?` into the orthography input field.
+  >   1. The parentheses around `e` aren't strictly necessary, but they help clarify what is being targeted by `?`.
```
Let's add parens as a character to know

jackrechsteiner replied:
Suggested parens use case: `pittsburgh(er)?`

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+
+What makes regular expressions more powerful than normal searches are metacharacters that have special functions.
+Some of the most useful metacharacters (and their functions) for searching APLS are:
+- `.` (matches any character)
```
I'm torn between intro-ing the metachars one by one vs. intro-ing idioms like `.*`. Perhaps we start with idioms, plop in a table of metachars, and replicate the table on the regex page?

djvill replied:
This is what I've used in LVC class in the past (with extensive boardwork and interactive trial-and-error): 
![image](https://github.com/user-attachments/assets/273db50a-5700-4ba8-bf78-9a51ea2948fe)

jackrechsteiner replied:
APLS characters that need to be escaped : `+` (morpheme boundaries), literal punctuation (`.` [short pause], `-` [long pause and syllable boundaries], `?` [question marker]), `{` and `$` (DISC vowels), also `$` is used in POS tags

jackrechsteiner replied:
This is the one thing I wasn't able to get to yet, but maybe getting some feedback on the current state of the Search page in the lab meeting tomorrow will be useful!

jackrechsteiner replied:
Maybe do a table intro-ing the metacharacters (includes link to other parts in the walkthrough and a warning that's like "If you've never done regex, start with the walkthrough) with some idioms below and then a walkthrough try-it below that. Maybe a 4th section for "If you've used regular expressions before..." (Or move the current note up to the top of the regex section so that experienced users see it right away)

djvill commented on doc/how-to-use/search.md:
```diff
@@ -5,28 +5,502 @@ parent: How to use APLS
+last_modified_date: 2025-03-28T20:52:49:z
 ---
 
 # {{ page.title }}
```
New organization!

- Searching the corpus (`/doc/search`)
  Content: Header text, What you can do (w/ examples), Single ortho try-it [maybe in the middle of What you can do], "more complicated searches will be covered in the pages in this section"
  - One word, one layer (`/doc/one-word-one-layer`): Maybe even just the ortho layer (in which case rename)
  - Search matches: But not exporting!
  - Participants and transcripts filters
  - Search options
  - (If OWOL is just ortho) Searching other layers: Intro-ing projects etc.
  - Data type (text, phono, numeric, mention timing-only just to link to anchoring)
  - Regex
  - Notation systems
  - Match/Doesn't match
  - Anchoring
  - Layer scope
  - Multiple words
  - Multiple layers
  - Multiple words and layers (maybe)
  - Tips and tricks
    A big basket of random helpful notes, like that you'll have to do some trial and error, refining your search via false positives and false negatives, maybe some troubleshooting
   Maybe in the form of an FAQ, maybe w/ a separate FAQ, maybe w/ a separate troubleshooting page

jackrechsteiner replied:
One change I've made wrt flow is putting "Data type" and "Notation systems" together – the inherent relationship between data types and notation systems made it feel choppy to have those sections broken up. 
Oh and I also put "Anchoring" after "Layer scope" (which is called "Searching segments within words" still)

jackrechsteiner committed:
Making some initial low-level wording edits and formatting changes

jackrechsteiner committed:
Adding in some examples with relative links in the "what you can do" section, reordering some try-its, and changing certain sentences to be in an active voice

jackrechsteiner committed:
Working on reorganizing the page to fit with Dan's suggested structure and adding in new sections where needed

jackrechsteiner committed:
Adding in lock icons

jackrechsteiner committed:
Updating the "Searching segments within words" section and sub-sections to have more exlpication of layer scopes

jackrechsteiner committed:
Finishing the reorganization and filling out the multi segment/word/layer sections

jackrechsteiner commented:
I haven't split the sections out into their own pages yet, but the revamped organization is now finished for the _search_ page documentation!

jackrechsteiner committed:
Updating relative links with new section names

jackrechsteiner committed:
Trying out a different structure for the regex section

jackrechsteiner committed:
Merge remote-tracking branch 'upstream/main'

jackrechsteiner committed:
The Great Search Schism of 2025 (where the single Search doc page is turned into multiple sub-pages and the relative links are updated throughout)

jackrechsteiner committed:
Attempting to add content to the regex page (but I'm having a bit of a hard time coming up with ideas)

jackrechsteiner committed:
Fixing typo

jackrechsteiner committed:
adding small comment note to main search page

jackrechsteiner committed:
Caught a typo in a relative link

jackrechsteiner committed:
Getting started on the "Exporting data" page

jackrechsteiner committed:
Removing try-it from search landing page and updating title of "Your first search" to "Your first search - Orthography"

jackrechsteiner committed:
Filling out the "Regex" page to be (mostly?) complete

djvill commented on doc/how-to-use/regex.md:
```diff
@@ -1,13 +1,231 @@
+
+The most important difference between regex in APLS and regex elsewhere, is that APLS (almost) always[^segment-exception] matches patterns to the **whole** word.
+
+[^segment-exception]: The only times that APLS does not look for matches by whole words is for the <span class="layer">segment</span> and <span class="layer">foll_segment</span> layers.
```
This is a good point. But/and it makes me realize that the larger principle at play is that LaBB-CAT matches patterns to the whole _annotation_[^label]. So if you're searching a layer whose annotation labels can have spaces (I think just comment/noise/redaction), then the pattern isn't matching a whole orthographic word. Also, if the layer has word scope and sub-interval alignment, the pattern can be for a _partial_ word.

[^label]: Technically, what we're matching isn't annotations, but their _labels_. But this is, I think, too fine a semantic distinction for end-users, so I've used phrasing like [notation systems are what those annotations [not annotation labels] actually _look like_](https://djvill.github.io/APLS/doc/layer-typology#notation-system). That said, the existence of timing-only layers forced me to hint at this distinction (e.g., [Phonological layers: Annotations **are labeled** with speech sounds](https://djvill.github.io/APLS/doc/layer-typology#data-type)), but I'm hoping that's too subtle for readers to notice.

jackrechsteiner committed:
I was getting some really odd merge conflicts with the `custom-entries.md` file, but I think (hope) I got it resolved

Merge remote-tracking branch 'upstream/main'

# Conflicts:
#	files/custom-dictionary/custom-entries.md

jackrechsteiner committed:
Regex page: fixing typos, making a subsection for inverted square brackets, clarifying that inverted square brackets still match a single character, changing the wording of the practice problems to be clearer, adding a "Tips and Tricks" section with a "Checking your search matches" subsection (and adding `book.svg` to use in that subsection)

jackrechsteiner committed:
Merge remote-tracking branch 'upstream/main'

jackrechsteiner committed:
Adding special chars to cheat sheet, turning escape chars sentence into table, updating invalid regex wording, changing APLS differences to be about annotations, and mentioning that dictionary export can be more useful than csv export in some cases

jackrechsteiner committed:
Finishing the content sections of the "Exporting data" documentation page

jackrechsteiner committed:
Starting on the content for the "Extracting acoustic measurements page"

jackrechsteiner committed:
Updating wording of _Dictionary Export_ section to reflect recent updates to the UI

jackrechsteiner committed:
Adding more content to "Extracting acoustic measurements" page

jackrechsteiner committed:
Deleting comments and fixing typo

jackrechsteiner committed:
Initial separation of "Process with Praat" and "Extracting additional data with a results .csv" doc pages

jackrechsteiner committed:
Finishing up the "Extract annotations for time intervals" section

jackrechsteiner committed:
Removing UI comments

jackrechsteiner committed:
Adding content to the "Extracting acoustic measurements" page
